src/sentry/workflow_engine/processors/workflow.py

```diff
@@ -109,6 +123,8 @@ def evaluate_workflows_action_filters(
             if evaluation:
                 filtered_action_groups.add(action_condition)
 
+        job.pop("workflow", None)
```

Comment:
maybe i don't need to pop, but not popping caused this problem in the first place